buildSrc/src/test/java/org/elasticsearch/gradle/EmptyDirTaskTests.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import static java.util.Collections.singletonMap;
+
+public class EmptyDirTaskTests extends GradleUnitTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    public void testCreateEmptyDir() throws Exception {
+        Project project = createProject();
+        EmptyDirTask emptyDirTask = createTask(project);
+        assertEquals(0755, emptyDirTask.getDirMode());
+
+        // generate a new temporary folder and make sure it does not exists
+        File newEmptyFolder = getNewTempFolderFile();
+
+        emptyDirTask.setDir(newEmptyFolder);
+        emptyDirTask.create();
+
+        assertEquals(true, newEmptyFolder.exists());
+        assertEquals(true, newEmptyFolder.isDirectory());
+        assertEquals(true, newEmptyFolder.canExecute());
+        assertEquals(true, newEmptyFolder.canRead());
+        assertEquals(true, newEmptyFolder.canWrite());
+    }
+
+    public void testCreateEmptyDirNoPermissions() throws Exception {
+        Project project = createProject();
+        EmptyDirTask emptyDirTask = createTask(project);
+        emptyDirTask.setDirMode(0000);
+
+        // generate a new temporary folder and make sure it does not exists
+        File newEmptyFolder = getNewTempFolderFile();
+
+        emptyDirTask.setDir(newEmptyFolder);
+        emptyDirTask.create();
+
+        assertEquals(true, newEmptyFolder.exists());
+        assertEquals(true, newEmptyFolder.isDirectory());
+        assertEquals(false, newEmptyFolder.canExecute());
+        assertEquals(false, newEmptyFolder.canRead());
+        assertEquals(false, newEmptyFolder.canWrite());
+    }
+
+    private File getNewTempFolderFile() throws IOException {
+        File newEmptyFolder = temporaryFolder.newFolder("newEmptyFolder");
+        newEmptyFolder.delete();
+        assertEquals(false, newEmptyFolder.exists());
+        return newEmptyFolder;
+    }
+
+    private Project createProject() throws IOException {
+        return ProjectBuilder.builder().withProjectDir(temporaryFolder.newFolder()).build();
+    }
+    private EmptyDirTask createTask(Project project) {
+        return (EmptyDirTask) project.task(singletonMap("type", EmptyDirTask.class), "emptyDirTask");
```

Comment:
I don't think we need to pass in a map like this, we can use `project.tasks.create` instead.